packages/wallets/brave/src/adapter.ts

```diff
@@ -0,0 +1,250 @@
+                this.emit('error', new WalletDisconnectionError(error?.message, error));
+            }
+        }
+
+        this.emit('disconnect');
+    }
+
+    async sendTransaction(
+        transaction: Transaction,
+        connection: Connection,
+        options?: SendTransactionOptions
+    ): Promise<TransactionSignature> {
+        try {
+            const wallet = this._wallet;
+            if (!wallet) throw new WalletNotConnectedError();
+
+            try {
+                // transaction.serializeMessage() requires feePayer and
+                // recentBlockhash to be set.
+                // Brave will set both here when caller doesn't set any of them.
+                // Otherwise, request would be rejected.
+                transaction.feePayer = transaction.feePayer || this.publicKey || undefined
+                transaction.recentBlockhash =
+                  transaction.recentBlockhash || (await connection.getRecentBlockhash('finalized')).blockhash;
+                const { signature } = await wallet.signAndSendTransaction(transaction, options);
+                return signature;
+            } catch (error: any) {
+                throw new WalletSendTransactionError(error?.message, error);
+            }
+        } catch (error: any) {
+            this.emit('error', error);
+            throw error;
+        }
+
+        return await super.sendTransaction(transaction, connection, options);
+    }
+
+    async signTransaction(transaction: Transaction): Promise<Transaction> {
+        try {
+            const wallet = this._wallet;
+            if (!wallet) throw new WalletNotConnectedError();
+
+            try {
+                return (await wallet.signTransaction(transaction)) || transaction;
+            } catch (error: any) {
+                throw new WalletSignTransactionError(error?.message, error);
+            }
+        } catch (error: any) {
+            this.emit('error', error);
+            throw error;
+        }
+    }
+
+    async signAllTransactions(transactions: Transaction[]): Promise<Transaction[]> {
+        try {
+            const wallet = this._wallet;
+            if (!wallet) throw new WalletNotConnectedError();
+
+            try {
+                return (await wallet.signAllTransactions(transactions)) || transactions;
+            } catch (error: any) {
+                throw new WalletSignTransactionError(error?.message, error);
+            }
+        } catch (error: any) {
+            this.emit('error', error);
+            throw error;
+        }
+    }
+
+    async signMessage(message: Uint8Array): Promise<Uint8Array> {
+        try {
+            const wallet = this._wallet;
+            if (!wallet) throw new WalletNotConnectedError();
+
+            try {
+                const { signature } = await wallet.signMessage(message);
+                return signature;
+            } catch (error: any) {
+                throw new WalletSignTransactionError(error?.message, error);
```

Comment:
I just caught this in several other adapters today, but this should be `WalletSignMessageError`

https://github.com/solana-labs/wallet-adapter/commit/112c54463a3d069e5b756dacf78e20a0711939ae fixes this elsewhere

Reply:
good call, fixed in https://github.com/solana-labs/wallet-adapter/pull/445/commits/b2b4e713bbe75bc8008a39829def9ce751745fc3